src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -26,17 +55,170 @@ const ActorInit Shot_Sun_InitVars = {
+    if (this->spawnTimer > 0) {
+        this->spawnTimer -= 1;
+        return;
+    }
+
+    switch (temp_v0) {
+        case 0x40:
+            sp38 = 7;
+            break;
+        case 0x41:
+            sp38 = 7;
+            break;
+    }
+
+    Actor_Spawn(
+        &globalCtx->actorCtx, globalCtx, ACTOR_EN_ELF,
+        this->actor.initPosRot.pos.x, this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z,
+        0, 0, 0,
+        sp38
+    );
+
+    Actor_Kill(&this->actor);
+}
+
+void ShotSun_StartFairyCountdown(ShotSun* this, GlobalContext* globalCtx) {
+    if ((func_8005B198() == this->actor.type) || (this->spawnTimer != 0)) {
+        this->actionFunc = &ShotSun_FairyCountdown;
+        this->spawnTimer = 50;
+
+        Actor_Spawn(
+            &globalCtx->actorCtx, globalCtx, ACTOR_DEMO_KANKYO,
+            this->actor.initPosRot.pos.x, this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z,
+            0, 0, 0,
+            0x11
+        );
+
+        func_80078914(&this->actor.projectedPos, NA_SE_EV_TRE_BOX_APPEAR);
+    }
+}
+
+void func_80BADF0C(ShotSun* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
+    s32 temp_t0 = this->actor.params & 0xFF;
+
+    if (22500.0f < func_800CB650(&this->actor.posRot.pos, &player->actor.posRot.pos)) {
+        this->unk_1A4[0] = 0;
+    } else {
+        if (this->unk_1A4[0] == 0) {
+            if (((s32)player->stateFlags2 << 7) >= 0) {
+                player->stateFlags2 |= 0x800000;
+                return;
+            } else {
+                this->unk_1A4[0] = 1;
+            }
+        }
+        if (this->unk_1A4[0] == 1) {
+            func_8010BD58(globalCtx, 1U);
+            this->unk_1A4[0] = 2;
+            return;
+        }
+        
+        if (this->unk_1A4[0] == 2 && globalCtx->msgCtx.unk_E3EE == 4) {
+            if (((temp_t0 == 0x40) && (*(u16*)(&globalCtx->msgCtx.unk_E3E8[4]) == 9))
+                || ((temp_t0 == 0x41) && (*(u16*)(&globalCtx->msgCtx.unk_E3E8[4]) == 0xB))) {
+                    this->actionFunc = &ShotSun_StartFairyCountdown;
+                    func_80080480(globalCtx, this);
+                    this->spawnTimer = 0;
+            } else {
+                this->unk_1A4[0] = 0;
+            }
+
+            this->unk_1A4[0] = 0;
+        }
+    }
+}
+
+void ShotSun_HyliaSunUpdate(ShotSun* this, GlobalContext* globalCtx) {
+    Vec3s cylinderPos;
+    Player* player = PLAYER;
+    EnItem00* collectible;
+    s32 pad;
+    Vec3f spawnPos;
+
+    if ((this->collider.base.acFlags & 2) != 0) {
+        func_80078884(NA_SE_SY_CORRECT_CHIME);
+        osSyncPrintf(VT_FGCOL(CYAN) "SHOT_SUN HIT!!!!!!!\n" VT_RST);
+        if (INV_CONTENT(SLOT_ARROW_FIRE) == ITEM_NONE) {
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ITEM_ETCETERA, 700.0f, -800.0f, 7261.0f, 0, 0, 0, 7);
+            globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&D_02007020);
+
+            // Maybe something was optimized away?
```

Comment:
you dont need to comment this, it happens all the time. you can also remove the whitespace around it